dashboard/app/models/script.rb

```diff
@@ -295,6 +295,36 @@ def self.get_from_cache(id)
     script_cache[id.to_s] || get_without_cache(id)
   end
 
+  def text_response_levels
+    return @text_response_levels if Script.should_cache? && @text_response_levels
+    @text_response_levels = text_response_levels_without_cache
+  end
+
+  def text_response_levels_without_cache
+    text_response_levels = []
+    script_levels.map do |script_level|
+      script_level.levels.map do |level|
+        unless level.contained_levels.empty?
+          text_response_levels << {
+            script_level: script_level,
+            levels: [level.contained_levels.first]
+          }
+        end
+      end
+    end
+
+    text_response_levels.concat(script_levels.includes(:levels).
+        where('levels.type' => [TextMatch, FreeResponse]).
```

Comment:
Ah I see this is existing code moved from above.  Maybe cleaner if these both inherited from a class that exposed `last_attempt` as `last_answer_text`, but no need to change this.